toolkit/analysis_and_visualization/classification/assets/scripts/pagination.js

```diff
@@ -0,0 +1,1125 @@
+        throwError("Pagination requires jQuery.");
+    }
+
+    var pluginName = "pagination";
+
+    var pluginHookMethod = "addHook";
+
+    var eventPrefix = "__pagination-";
+
+    // Conflict, use backup
+    if ($.fn.pagination) {
+        pluginName = "pagination2";
+    }
+
+    $.fn[pluginName] = function (options) {
+
+        if (typeof options === "undefined") {
+            return this;
+        }
+
+        var container = $(this);
+
+        var attributes = $.extend({}, $.fn[pluginName].defaults, options);
+
+        var pagination = {
+
+            initialize: function () {
+                var self = this;
+
+                // Cache attributes of current instance
+                if (!container.data("pagination")) {
+                    container.data("pagination", {});
+                }
+
+                if (self.callHook("beforeInit") === false) return;
+
+                // Pagination has been initialized, destroy it
+                if (container.data("pagination").initialized) {
+                    $(".paginationjs", container).remove();
+                }
+
+                // Whether to disable Pagination at the initialization
+                self.disabled = !!attributes.disabled;
+
+                // Model will be passed to the callback function
+                var model = self.model = {
+                    pageRange: attributes.pageRange,
+                    pageSize: attributes.pageSize
+                };
+
+                // dataSource`s type is unknown, parse it to find true data
+                self.parseDataSource(attributes.dataSource, function (dataSource) {
+
+                    // Currently in asynchronous mode
+                    self.isAsync = Helpers.isString(dataSource);
+                    if (Helpers.isArray(dataSource)) {
+                        model.totalNumber = attributes.totalNumber = dataSource.length;
+                    }
+
+                    // Currently in asynchronous mode and a totalNumberLocator is specified
+                    self.isDynamicTotalNumber = self.isAsync && attributes.totalNumberLocator;
+
+                    var el = self.render(true);
+
+                    // Add extra className to the pagination element
+                    if (attributes.className) {
+                        el.addClass(attributes.className);
+                    }
+
+                    model.el = el;
+
+                    // Append/prepend pagination element to the container
+                    container[attributes.position === "bottom" ? "append" : "prepend"](el);
+
+                    // Bind events
+                    self.observer();
+
+                    // Pagination is currently initialized
+                    container.data("pagination").initialized = true;
+
+                    // Will be invoked after initialized
+                    self.callHook("afterInit", el);
+                });
+            },
+
+            render: function (isBoot) {
```

Comment:
Codacy found an issue: [Method 'render' has a complexity of 9. Maximum allowed is 4.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6397592)